vignettes/icommands.Rmd

```diff
@@ -0,0 +1,203 @@
+
+`{rirods}` | `iCommands` | Result
+----- | ----- | ----------
+`iget("myfile.txt", "myfile.txt")` | `iget myfile.txt` | "myfile.txt" is copied from iRODS to local as "myfile.txt"
+`iget("myfile.txt", "another_path/filename.txt")` | `iget myfile.txt another_path/filename.txt` | "myfile.txt" is copied from iRODS to local as "filename.txt" in "another_path"
+`iput("data.csv", "data.csv")` | `iput data.csv` | "data.csv" is copied from local to iRODS as "data.csv"
+`iput("data.csv", "data2.csv")` | `iput data.csv data2.csv` | "data.csv" is copied from local to iRODS as "data2.csv" 
+
+Next to these functions, `{rirods}` provides two functions to store R objects directly into iRODS or read RDS files directly from iRODS: `isaveRDS()` and `ireadRDS()`. There is no counterpart in iCommands. To achieve the same effect as `isaveRSD()`, a user would first have to save their object with `saveRDS()`, then send it to iRODS with `iput` and finally remove it with `irm -f`; in the same vein, the effect of `ireadRDS()` would be achieved by retrieving the file with `iget`, reading it with `readRDS()` and then removing it from the local space.
+Read more about this in `vignette("local-irods")`.
+
+## Metadata and querying
+
+The `{rirods}` `imeta()` function covers two of the main functionalities of the `imeta` iCommand: adding and removing metadata from a data object or collection. This function has three main arguments: the logical path of the data object or collection on which to add the metadata, the entity type (e.g. "data_object", the default, or "collection") and a list of operations, in which it is specified whether a certain AVU should be added or removed.
+
+```{r}
+icd("some_collection")
+ils(metadata=TRUE)
+
+# imeta add -C subcollection foo bar baz
+imeta("subcollection", "collection",
+      operations = list(
+        list(operation="add", attribute="foo", value="bar", units="baz")
+        )
+      )
+
+# imeta rm -d random_numbers.rds distribution normal
+imeta("random_numbers.rds",
+      operations = list(
+        list(operation="remove", attribute="distribution", value="normal")
+      ))
+
+ils(metadata=TRUE)
+```
+
+In order to "modify" an AVU with `{rirods}`, we would have to remove it and add its replacement. This also illustrates how we can add/remove several AVUs in one `imeta()` call, which would not be possible with `iCommands`.
```

Comment:
"with `iCommands`" -> "with `imeta`"

Reply:
I can change it, but there is no way of adding or removing several AVUs in one call with `iCommands`, though, right? Or am I missing something?

(I've seen people use loops to add AVUs via the PRC instead of using the atomic operations, so I thought it would be useful to emphasize the possibility of sending a list of AVUs directly for users that only know how to add one AVU at a time.)

Reply:
Oh, I understand now - I was just trying to be more specific.

That said, yes, `imeta` can only send one at a time.

However, you can set multiple AVUs at once with `iput --metadata`.

https://docs.irods.org/4.3.0/icommands/user/#iput

```
 --metadata - atomically assign metadata after a data object is registered in
              the catalog. Metadata is encoded into a quoted string of the
              form attr1;val1;unit1;attr2;val2;unit2;
```